src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/keyColumnSorting.ts

```diff
@@ -0,0 +1,230 @@
+  let temp = sortKeysArrayByName(keys);
+
+  if (!sortDirection) {
+    temp = temp.reverse();
+  }
+
+  temp.forEach((key: Key) => {
+    if (key.certificates) {
+      sortDirection
+        ? (key.certificates = sortCertsByNameAsc(key.certificates))
+        : (key.certificates = sortCertsByNameAsc(key.certificates).reverse());
+    }
+  });
+
+  return temp;
+};
+
+/**
+ * Sort certificates by name Ascending
+ */
+export const sortCertsByNameAsc = (
+  certs: TokenCertificate[],
+): TokenCertificate[] => {
+  return certs.sort((a: TokenCertificate, b: TokenCertificate) => {
+    return (
+      a.certificate_details.issuer_common_name + a.certificate_details.serial
+    ).localeCompare(
+      b.certificate_details.issuer_common_name + b.certificate_details.serial,
+    );
+  });
+};
+
+/**
+ * Sort certificates using given functions
+ */
+export const sortCertificatesForKeys = (
+  keys: Key[],
+  sortDirection: boolean,
+  sortingValue: string,
+): Key[] => {
+  const temp = keys;
+
+  temp.forEach((key: Key) => {
+    if (key.certificates) {
+      if (sortDirection) {
+        key.certificates = sortCertsAsc(key.certificates, sortingValue);
+      } else {
+        key.certificates = sortCertsAsc(
+          key.certificates,
+          sortingValue,
+        ).reverse();
+      }
+    }
+  });
+
+  return temp;
+};
+
+/**
+ * Sort certificates by date
+ */
+export const sortKeysByDate = (keys: Key[], sortDirection: boolean): Key[] => {
+  const temp = keys;
+
+  temp.forEach((key: Key) => {
+    if (key.certificates) {
+      if (sortDirection) {
+        key.certificates = sortCertsByDateAsc(key.certificates);
+      } else {
+        key.certificates = sortCertsByDateAsc(key.certificates).reverse();
+      }
+    }
+  });
+
+  return temp;
+};
+
+export const sortCertsAsc = (
+  certs: TokenCertificate[],
+  sortingValue: string,
+): TokenCertificate[] => {
+  return certs.sort((a: TokenCertificate, b: TokenCertificate) => {
+    return (a as any)[sortingValue].localeCompare((b as any)[sortingValue]);
+  });
+};
+
+/**
+ * Sort certificates by date Ascending
+ */
+export const sortCertsByDateAsc = (
+  certs: TokenCertificate[],
+): TokenCertificate[] => {
+  return certs.sort((a: TokenCertificate, b: TokenCertificate) => {
+    return a.certificate_details.not_after.localeCompare(
+      b.certificate_details.not_after,
+    );
+  });
+};
+
+/**
+ * Sort keys including certificates and csr:s by id
+ */
+export const sortKeysById = (keys: Key[], sortDirection: boolean): Key[] => {
+  const temp = sortCertificatesForKeys(keys, sortDirection, 'owner_id');
+
+  temp.forEach((key: Key) => {
+    if (key.certificate_signing_requests) {
+      if (sortDirection) {
+        key.certificate_signing_requests = sortRequestsAsc(
+          key.certificate_signing_requests,
+        );
+      } else {
+        key.certificate_signing_requests = sortRequestsAsc(
+          key.certificate_signing_requests,
+        ).reverse();
```

Comment:
I think this could also be simplified to this:

`
const sortedRequests = sortRequestsAsc(key.certificate_signing_requests);
key.certificate_signing_requests = sortDirection ? sortedRequests : sortedRequests.reverse();
`

Reply:
Fixed